airbyte-integrations/base-java/src/main/java/io/airbyte/integrations/javabase/IntegrationRunner.java

```diff
@@ -0,0 +1,117 @@
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.Scanner;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class IntegrationRunner {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(IntegrationRunner.class);
+
+  private final Destination destination;
+
+  public IntegrationRunner(Destination destination) {
+    this.destination = destination;
+  }
+
+  private void run(String[] args) throws Exception {
+    final IntegrationConfig parsed = new IntegrationCliParser().parse(args);
+
+    switch (parsed.getCommand()) {
+      case SPEC -> System.out.println(Jsons.serialize(destination.spec()));
+      case CHECK -> {
+        final JsonNode config = parseConfig(parsed.getConfig());
+        System.out.println(Jsons.serialize(destination.check(config)));
+      }
+      case DISCOVER -> {
+        final JsonNode config = parseConfig(parsed.getConfig());
+        System.out.println(Jsons.serialize(destination.discover(config)));
+      }
+      case READ ->
+        // final JsonNode config = parseConfig(parsed.getConfig());
+        // final Schema schema = parseConfig(parsed.getSchema(), Schema.class);
+        // final State state = parseConfig(parsed.getState(), State.class);
+        throw new RuntimeException("Not implemented");
+      case WRITE -> {
+        final JsonNode config = parseConfig(parsed.getConfig());
+        final Schema schema = parseConfig(parsed.getSchema(), Schema.class);
+        final DestinationConsumer<SingerMessage> consumer = destination.write(config, schema);
+
+        final Scanner input = new Scanner(System.in);
+        try (consumer) {
+          while (input.hasNextLine()) {
+            final Optional<SingerMessage> singerMessageOptional = Jsons.tryDeserialize(input.nextLine(), SingerMessage.class);
+            if (singerMessageOptional.isPresent()) {
+              consumer.accept(singerMessageOptional.get());
+            }
+          }
+          consumer.complete();
+        }
+      }
+      default -> throw new IllegalStateException("Unexpected value: " + parsed.getCommand());
+    }
+  }
+
+  private static JsonNode parseConfig(String path) {
+    return Jsons.deserialize(IOs.readFile(Path.of(path)));
+  }
+
+  private static <T> T parseConfig(String path, Class<T> klass) {
+    final JsonNode jsonNode = parseConfig(path);
+    return Jsons.object(jsonNode, klass);
+  }
+
+  public static void main(String[] args) {
+    try {
+
+      final String destinationClass = System.getenv().get(JavaBaseConstants.ENV_DESTINATION_CLASS);
+      final String destinationJarPath = System.getenv().get(JavaBaseConstants.ENV_DESTINATION_JAR_PATH);
+
+      LOGGER.info("destination class: {}", destinationClass);
+      LOGGER.info("destination jar path: {}", destinationJarPath);
+
+      // add the specific destination to the classpath.
+      final URLClassLoader urlClassLoader = new URLClassLoader(new URL[] {Path.of(destinationJarPath).toUri().toURL()});
```

Comment:
How is it going to work if the destination code has dependencies?

Reply:
discussed offline, agreed to just have destination have the main method which obviates all of this.